crates/ra_hir/src/code_model_impl/module.rs

```diff
@@ -126,15 +131,39 @@ impl Module {
         );
 
         let segments = &path.segments;
-        for name in segments.iter() {
+        for (idx, name) in segments.iter().enumerate() {
             let curr = if let Some(r) = curr_per_ns.as_ref().take_types() {
                 r
             } else {
                 return Ok(PerNs::none());
             };
             let module = match curr.resolve(db)? {
                 Def::Module(it) => it,
-                // TODO here would be the place to handle enum variants...
+                Def::Enum(e) => {
+                    if segments.len() == idx + 1 {
+                        // enum variant
+                        let matching_variant = e
+                            .variant_ids()
+                            .iter()
+                            .filter_map(|&variant_id| {
+                                EnumVariant::new(variant_id, e.def_id())
+                                    .name(db)
+                                    .ok()
+                                    .and_then(|variant_name| variant_name.map(|n| (variant_id, n)))
+                            })
+                            .find(|(_variant_id, variant_name)| variant_name == name);
+
+                        if let Some(variant) = matching_variant {
+                            return Ok(PerNs::both(variant.0, e.def_id()));
+                        } else {
+                            return Ok(PerNs::none());
+                        }
+                    } else {
+                        // enum
+                        assert_eq!(segments.len(), idx);
```

Comment:
this should not assert -- it's not a bug in rust-analyzer if this happens. I can easily write such a path, and it would be quite bad if we just crashed when trying to resolve that. We should instead just return no resolution.

Reply:
fixed!